perfkitbenchmarker/linux_benchmarks/object_storage_service_benchmark.py

```diff
@@ -346,48 +355,105 @@ def _ProcessMultiStreamResults(raw_result, operation, sizes,
     sizes: the object sizes used in the benchmark, in bytes.
     results: a list to append Sample objects to.
     metadata: dict. Base sample metadata
-
   """
 
+  num_streams = FLAGS.object_storage_multistream_num_streams
+
   if metadata is None:
     metadata = {}
-
-  records = json.loads(raw_result)
-  metric_name = 'Multi-stream %s latency' % operation
-
-  metadata = metadata.copy()
-  metadata['num_streams'] = FLAGS.object_storage_multistream_num_streams
+  metadata['num_streams'] = num_streams
   metadata['objects_per_stream'] = (
       FLAGS.object_storage_multistream_objects_per_stream)
 
-  overall_metadata = metadata.copy()
+  records_json = json.loads(raw_result)
+  records = pd.DataFrame(records_json)
+
+  all_streams_start, all_streams_duration = analyze.AllStreamsInterval(
+      records['start_time'], records['latency'], records['stream_num'])
+  start_gap, stop_gap = analyze.StreamStartAndEndGaps(
+      records['start_time'], records['latency'],
+      all_streams_start, all_streams_duration)
+  if ((start_gap + stop_gap) / all_streams_duration <
+      MULTISTREAM_STREAM_GAP_THRESHOLD):
+    logging.info(
+        'First stream started %s seconds before last stream started', start_gap)
+    logging.info(
+        'Last stream ended %s seconds after first stream ended', stop_gap)
+  else:
+    logging.warning(
+        'Difference between first and last stream start/end times was %s and '
+        '%s, which is more than %s of the benchmark time %s.',
+        start_gap, stop_gap, MULTISTREAM_STREAM_GAP_THRESHOLD,
+        all_streams_duration)
+    metadata['stream_gap_above_threshold'] = True
+
+  records_in_interval = records[
+      analyze.FullyInInterval(records['start_time'],
+                              records['latency'],
+                              all_streams_start,
+                              all_streams_duration)]
+
   # Don't publish the full distribution in the metadata because doing
   # so might break regexp-based parsers that assume that all metadata
   # values are simple Python objects. However, do add an
   # 'object_size_B' metadata field even for the full results because
   # searching metadata is easier when all records with the same metric
   # name have the same set of metadata fields.
-  overall_metadata['object_size_B'] = 'distribution'
+  distribution_metadata = metadata.copy()
+  distribution_metadata['object_size_B'] = 'distribution'
+
+  latency_prefix = 'Multi-stream %s latency ' % operation
   logging.info('Processing %s multi-stream %s results for the full '
-               'distribution.', len(records), operation)
-  _AppendPercentilesToResults(
-      results,
-      (record['latency'] for record in records),
-      metric_name,
+               'distribution.', len(records_in_interval), operation)
+  analyze.AppendStatsAsSamples(
+      records_in_interval['latency'],
```

Comment:
For latency percentiles and results reporting, I think we should just keep it simple and report all latency data as is, there is no need to go through this interval selection process which is only necessary for aggregated throughput calculation.